### ISSUE TITLE
audio load error in GigaSpeech recipe

### ISSUE THREAD

**Issue by csukuangfj** [lhotse-speech/lhotse#489]
See the error log below:

```
ValueError: Requested more audio (42213.59s) than available (42213.5886875s)
```

---

```
2021-11-29 10:12:31,581 INFO [compute_fbank_gigaspeech_splits.py:93] Loading data/fbank/XL_split_1000/cuts_XL_raw.0180.jsonl.gz
2021-11-29 10:12:31,923 INFO [compute_fbank_gigaspeech_splits.py:96] Computing features
Computing features in batches:  99%|________________________________________________________________________________| 113/114 [05:15<00:02,  2.80s/it]
Traceback (most recent call last):
  File "./local/compute_fbank_gigaspeech_splits.py", line 146, in <module>
    main()
  File "./local/compute_fbank_gigaspeech_splits.py", line 142, in main
    compute_fbank_gigaspeech_splits(args)
  File "./local/compute_fbank_gigaspeech_splits.py", line 98, in compute_fbank_gigaspeech_splits
    cut_set = cut_set.compute_and_store_features_batch(
  File "/ceph-fj/fangjun/open-source-2/lhotse-giga/lhotse/cut.py", line 3819, in compute_and_store_features_batch
    for batch in dloader:
  File "/ceph-fj/fangjun/py38/lib/python3.8/site-packages/torch/utils/data/dataloader.py", line 435, in __next__
    data = self._next_data()
  File "/ceph-fj/fangjun/py38/lib/python3.8/site-packages/torch/utils/data/dataloader.py", line 1085, in _next_data
    return self._process_data(data)
  File "/ceph-fj/fangjun/py38/lib/python3.8/site-packages/torch/utils/data/dataloader.py", line 1111, in _process_data
    data.reraise()
  File "/ceph-fj/fangjun/py38/lib/python3.8/site-packages/torch/_utils.py", line 428, in reraise
    raise self.exc_type(msg)
ValueError: Caught ValueError in DataLoader worker process 11.
Original Traceback (most recent call last):
  File "/ceph-fj/fangjun/open-source-2/lhotse-giga/lhotse/utils.py", line 587, in wrapper
    return fn(*args, **kwargs)
  File "/ceph-fj/fangjun/open-source-2/lhotse-giga/lhotse/audio.py", line 354, in load_audio
    samples = source.load_audio(
  File "/ceph-fj/fangjun/open-source-2/lhotse-giga/lhotse/audio.py", line 152, in load_audio
    raise ValueError(
ValueError: Requested more audio (42213.59s) than available (42213.5886875s)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/ceph-fj/fangjun/open-source-2/lhotse-giga/lhotse/utils.py", line 587, in wrapper
    return fn(*args, **kwargs)
  File "/ceph-fj/fangjun/open-source-2/lhotse-giga/lhotse/cut.py", line 952, in load_audio
    return self.recording.load_audio(
  File "/ceph-fj/fangjun/open-source-2/lhotse-giga/lhotse/utils.py", line 589, in wrapper
    raise type(e)(
ValueError: Requested more audio (42213.59s) than available (42213.5886875s)
[extra info] When calling: Recording.load_audio(args=(Recording(id='YOU0000004887', sources=[AudioSource(type='file', channels=[0], s
ource='/ceph-fj/fangjun/open-source-2/icefall-giga2/egs/gigaspeech/ASR/download/GigaSpeech/audio/youtube/P0049/YOU0000004887.opus')],
 sampling_rate=16000, num_samples=675417440, duration=42213.59, transforms=None),) kwargs={'channels': 0, 'offset': 0, 'duration': 42
213.59})

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/ceph-fj/fangjun/py38/lib/python3.8/site-packages/torch/utils/data/_utils/worker.py", line 198, in _worker_loop
    data = fetcher.fetch(index)
  File "/ceph-fj/fangjun/py38/lib/python3.8/site-packages/torch/utils/data/_utils/fetch.py", line 46, in fetch
    data = self.dataset[possibly_batched_index]
  File "/ceph-fj/fangjun/open-source-2/lhotse-giga/lhotse/dataset/unsupervised.py", line 70, in __getitem__
    return {"cuts": cuts, "audio": [c.load_audio() for c in cuts]}
  File "/ceph-fj/fangjun/open-source-2/lhotse-giga/lhotse/dataset/unsupervised.py", line 70, in <listcomp>
    return {"cuts": cuts, "audio": [c.load_audio() for c in cuts]}
  File "/ceph-fj/fangjun/open-source-2/lhotse-giga/lhotse/utils.py", line 589, in wrapper
    raise type(e)(
ValueError: Requested more audio (42213.59s) than available (42213.5886875s)

[extra info] When calling: Recording.load_audio(args=(Recording(id='YOU0000004887', sources=[AudioSource(type='file', channels=[0], source='/ceph-fj/fangjun/open-source-2/icefallgiga2/egs/gigaspeech/ASR/download/GigaSpeech/audio/youtube/P0049/YOU0000004887.opus')],
 sampling_rate=16000, num_samples=675417440, duration=42213.59, transforms=None),) kwargs={'channels': 0, 'offset': 0, 'duration': 42213.59})
[extra info] When calling: MonoCut.load_audio(args=(MonoCut(id='YOU0000004887-20519-0', start=0, duration=42213.59, channel=0, superv
isions=[SupervisionSegment(id='YOU0000004887_S0000127', recording_id='YOU0000004887', start=6099.69, duration=0.9, channel=0, text='T
O YOU', language='English', speaker='N/A', gender=None, custom=None, alignment=None), SupervisionSegment(id='YOU0000004887_S0000138',
 recording_id='YOU0000004887', start=6192.73, duration=4.44, channel=0, text='WITHOUT OBJECTION SO ORDERED THE START AND AND ARMS WIL
L MAKE THE PROCLAMATION', language='English', speaker='N/A', gender=None, custom=None, alignment=None), SupervisionSegment(id='YOU000
0004887_S0000141', recording_id='YOU0000004887', start=6223.91, duration=2.1, channel=0, text='THE MAJORITY LEADERS RECOGNIZE', langu
age='English', speaker='N/A', gender=None, custom=None, alignment=None), SupervisionSegment(id='YOU0000004887_S0000144', recording_id
='YOU0000004887', start=6253.0, duration=2.31, channel=0, text='MR CHIEF JUSTICE', language='English', speaker='N/A', gender=None, cu
stom=None, alignment=None), SupervisionSegment(id='YOU0000004887_S0000146', recording_id='YOU0000004887', start=6266.35, duration=6.0
9, channel=0, text='ON BEHALF OF MYSELF SENATOR MURKOWSKI AND SENATOR FROM', language='English', speaker='N/A', gender=None, custom=N
one, alignment=None), SupervisionSegment(id='YOU0000004887_S0000155', recording_id='YOU0000004887', start=6604.11, duration=2.04, cha
nnel=0, text='THE THE MAJORITY LEADERS RECOGNIZED', language='English', speaker='N/A', gender=None, custom=None, alignment=None), Sup
ervisionSegment(id='YOU0000004887_S0000174', recording_id='YOU0000004887', start=6882.92, duration=15.78, channel=0, text="REALLY THE
IR GOAL SHOULD BE TO GIVE YOU ALL THE FACTS BECAUSE THEY'RE ASKING YOU TO DO SOMETHING VERY VERY CONSEQUENTIAL AND ASK YOURSELF ASK Y
OURSELF", language='English', speaker='N/A', gender=None, custom=None, alignment=None), SupervisionSegment(id='YOU0000004887_S0000175
', recording_id='YOU0000004887', start=6900.08, duration=3.45, channel=0, text="GIVEN THE FACTS YOU HEARD TODAY THAT THEY DIDN'T TELL
 YOU", language='English', speaker='N/A', gender=None, custom=None, alignment=None), SupervisionSegment(id='YOU0000004887_S0000178',
recording_id='YOU0000004887', start=6917.94, duration=1.74, channel=0, text='THEY SHOULD GIVE YOU THE FACTS', language='English', spe
aker='N/A', gender=None, custom=None, alignment=None), SupervisionSegment(id='YOU0000004887_S0000180', recording_id='YOU0000004887',
start=6927.21, duration=1.02, channel=0, text='NUMBER THREE', language='English', speaker='N/A', gender=None, custom=None, alignment=
None), SupervisionSegment(id='YOU0000004887_S0000182', recording_id='YOU0000004887', start=6934.3, duration=9.57, channel=0, text='NO
T A SINGLE WITNESS IN THE HOUSE RECORD THAT THEY COMPILED IN DEVELOPED UNDER THEIR PROCEDURES THAT WE DISCUSSED AND WILL CONTINUE TO
DISCUSS', language='English', speaker='N/A', gender=None, custom=None, alignment=None), SupervisionSegment(id='YOU0000004887_S0000198
', recording_id='YOU0000004887', start=7307.83, duration=1.77, channel=0, text='THE SENATOR FROM MASSACHUSETTS', language='English',
speaker='N/A', gender=None, custom=None, alignment=None), SupervisionSegment(id='YOU0000004887_S0000199', recording_id='YOU0000004887
', start=7310.74, duration=1.98, channel=0, text='SEND A QUESTION TO THE BEST', language='English', speaker='N/A', gender=None, custo
m=None, alignment=None), SupervisionSegment(id='YOU0000004887_S0000208', recording_id='YOU0000004887', start=7400.12, duration=11.16,
 channel=0, text='AH WHEN WE RAISE THE SUBPOENA WITH JOHN BOLES COUNCIL THE SAME COUNCIL FOUR DR KOPPELMAN THE ANSWER WAS SENATOR YOU
 SERVICE FOR THE SUBPOENA AND WE WILL SUE U TWO ', language='English', speaker='N/A', gender=None, custom=None, alignment=None), Supe
rvisionSegment(id='YOU0000004887_S0000222', recording_id='YOU0000004887', start=7669.23, duration=1.62, channel=0, text='MR CHIEF JUS
TICE', language='English', speaker='N/A', gender=None, custom=None, alignment=None), SupervisionSegment(id='YOU0000004887_S0000226',
recording_id='YOU0000004887', start=7741.98, duration=2.37, channel=0, text='MR CHIEF JUSTICE CENTERS', language='English', speaker='
N/A', gender=None, custom=None, alignment=None), SupervisionSegment(id='YOU0000004887_S0000242', recording_id='YOU0000004887', start=
8139.37, duration=1.05, channel=0, text='THIRD', language='English', speaker='N/A', gender=None, custom=None, alignment=None), Superv
isionSegment(id='YOU0000004887_S0000246', recording_id='YOU0000004887', start=8204.52, duration=1.42, channel=0, text='AND AMID ONE F
INAL POINT', language='English', speaker='N/A', gender=None, custom=None, alignment=None), SupervisionSegment(id='YOU0000004887_S0000
247', recording_id='YOU0000004887', start=8207.5, duration=13.17, channel=0, text="AGAIN IF YOU HAVE ANY LINGERING QUESTIONS ABOUT DI
RECT EVIDENCE ANY THOUGHTS ABOUT ANYTHING WE JUST TALKED ABOUT ANYTHING I'D JUST RELAYED OR THAT WE'VE TALKED ABOUT THE LAST WEEK THE
RE IS A WAY TO SHED ADDITIONAL LIGHT ON", language='English', speaker='N/A', gender=None, custom=None, alignment=None), SupervisionSe
gment(id='YOU0000004887_S0000252', recording_id='YOU0000004887', start=8259.81, duration=2.07, channel=0, text='OF COUNSEL FOR THE PR
ESIDENT', language='English', speaker='N/A', gender=None, custom=None, alignment=None), SupervisionSegment(id='YOU0000004887_S0000256
', recording_id='YOU0000004887', start=8368.81, duration=2.79, channel=0, text='IN FOREIGN AFFAIRS TO SOME EXTENT', language='English
', speaker='N/A', gender=None, custom=None, alignment=None), SupervisionSegment(id='YOU0000004887_S0000261', recording_id='YOU0000004
887', start=8568.15, duration=2.85, channel=0, text='CENTER SHE YEAH I MEAN ASKS THE HOUSE MANAGERS', language='English', speaker='N/
A', gender=None, custom=None, alignment=None), SupervisionSegment(id='YOU0000004887_S0000275', recording_id='YOU0000004887', start=90
87.06, duration=5.01, channel=0, text='IN ORDER TO ALLOW US TO PRESENT THE TRUTH TO THE AMERICAN PEOPLE', language='English', speaker
='N/A', gender=None, custom=None, alignment=None), SupervisionSegment(id='YOU0000004887_S0000276', recording_id='YOU0000004887', star
t=9092.91, duration=7.83, channel=0, text='IN THE OCTOBER EIGHT LETTER THAT WAS SENT TO THE HOUSE OF REPRESENTATIVES THERE WAS NO', l
anguage='English', speaker='N/A', gender=None, custom=None, alignment=None), SupervisionSegment(id='YOU0000004887_S0000279', recordin
g_id='YOU0000004887', start=9143.99, duration=2.07, channel=0, text='MR CHIEF JUSTICE SENATORS ', language='English', speaker='N/A',
... ... ...
```

**Comment by csukuangfj** [comment]
The  problematic utterance in the cuts file is 
<img width="1337" alt="Screen Shot 2021-11-29 at 10 38 17 AM" src="https://user-images.githubusercontent.com/5284924/143800113-20d9987a-c899-4e96-9cf7-71968d7e90aa.png">

From the error log:
> ValueError: Requested more audio (42213.59s) than available (42213.5886875s)

Maybe the duration in the cuts file is rounded up from `.58886875` to `.59` but the code does not handle the round-off error.

**Comment by pzelasko** [comment]
Please see if https://github.com/lhotse-speech/lhotse/pull/491 helps solve your issue
